### ISSUE TITLE
Cant generate typescript interfaces from api-platform

### ISSUE THREAD

**Issue by misaon** [Sylius/Sylius#11562]
**Sylius version affected**: latest

**Description**  
I found a bug where api-platform could not export typescript interfaces from "new-api".
`Error: Unable to find the URL for "https://master.demo.sylius.com/new-api/docs.jsonld#Entrypoint/registerShopUser", make sure your api resource has at least one GET item operation declared.`

**Steps to reproduce**  
Just run: `$ npx @api-platform/client-generator --generator typescript https://master.demo.sylius.com/new-api test/`

**Possible Solution**  
Add get collection operation to your registerShopUser resource.

**Comment by misaon** [comment]
Some update for this?

**Comment by sergerdn** [comment]
Hi, @misaon. I found the same bug in another project based on api-platform. But my experience very low. Can I ask you to create new issue  in https://github.com/api-platform/client-generator? I belive that it should be fixed in their side.

**Comment by LeoMoshko** [comment]
Hi @misaon, Has the problem been resolved? I have a similar case:

`Error: Unable to find the URL for "https://127.0.0.1:8000/new-api/docs.jsonld#Entrypoint/avatarImage", make sure your api resource has at least one GET item operation declared.`

**Comment by johnkhansrc** [comment]
Hi
Same problem.

We want to generate TS interfaces from swagger.
15% of our API is made up of custom API endpoints who represent other ressources this isen't no related to entities.

Ts generator seem to be blocked on the first custom API endpoint.

Custom API endpoints are defined like this :
```php
namespace App\SwaggerEndpoint;

use ApiPlatform\Core\Annotation\ApiResource;

/**
 * @ApiResource(
 *     collectionOperations={
 *         "get": {
 *             "route_name": "app_v2_ads_getadbyid",
 *             "method": "GET",
 *             "openapi_context": {
 *                 "summary": "Get ads webview",
 *             },
 *         },
 *     },
 *     itemOperations={}
 * )
 */
class Ads
{
}

```


![Capture d’écran du 2021-03-29 18-39-25](https://user-images.githubusercontent.com/28094259/112870271-52aa2380-90be-11eb-9ac9-4768ef3790ad.png)


We execute that command :
`NODE_TLS_REJECT_UNAUTHORIZED=0 npx @api-platform/client-generator --generator typescript https://xxxxxxxxx/ tsI/`
This return :
`Error: Unable to find the URL for "https://xxxxxxxxx/docs.jsonld#Entrypoint/ads", make sure your api resource has at least one GET item operation declared.`

Is it possible to work around this problem  ?

Best regards.

**Comment by misaon** [comment]
@lchrusciel https://github.com/api-platform/client-generator/issues/229#issuecomment-712810498

What do you think about it?

**Comment by dantleech** [comment]
A dirty fix is to edit:

**Comment by dantleech** [comment]
hmm, I notice the generator seems not to generate the Shop types, only `product.ts` for example

**Comment by Tcharlyson** [comment]
Any news about this ?

**Comment by lchrusciel** [comment]
hey folks, I'm aware that you may be pissed off about my long response time, sorry about that. However, client generation was not our main goal at this stage. As we are finishing with the shop part, there are chances to finally fix it (probably this summer).

I have to check our API design and ensure that there always is GET for resource 👍

**Comment by SalvadorCardona** [comment]
I'm use other library for generated a typescript interface, i'm use npx openapi-typescrip with this command : 

```
npx openapi-typescript npx openapi-typescript https://master.demo.sylius.com/api/v2/docs.json --output ./src/schema/app-api-schema.ts
```

After i'm use the library openapi-typescript-fetch for autocomplete all route api in typescript like this

<img width="685" alt="image" src="https://user-images.githubusercontent.com/18017015/173027048-f0deddff-71e7-456b-9fbe-7f5ca850cd54.png">

**Comment by misaon** [comment]
> I'm use other library for generated a typescript interface, i'm use npx openapi-typescrip with this command :
> 
> ```
> npx openapi-typescript npx openapi-typescript https://master.demo.sylius.com/api/v2/docs.json --output ./src/schema/app-api-schema.ts
> ```
> 
> After i'm use the library openapi-typescript-fetch for autocomplete all route api in typescript like this
> 
> <img alt="image" width="685" src="https://user-images.githubusercontent.com/18017015/173027048-f0deddff-71e7-456b-9fbe-7f5ca850cd54.png">

`npx openapi-typescript https://master.demo.sylius.com/api/v2/docs.json --output ./src/schema/app-api-schema.ts` works great thanks!
